toy_2d/src/ilqr.py

```diff
@@ -161,37 +159,42 @@ def forward_pass(self, xx, uu, dd, KK, lr):
         # TODO: compute forward pass
         for i in range(self.N-1):
             utraj[i] = uu[i] + KK[i]@(xtraj[i] - xx[i]) + lr*dd[i]
+            # if(utraj[i][0]<0 and utraj[i][1]>0): utraj[i] = np.array([0,0])
+            #TO DO
+            # xtraj[i+1] = quad_sim.F(xtraj[i], utraj[i], self.dt)
             xtraj[i+1] = self.sim_forward(xtraj[i], utraj[i])
         return xtraj, utraj
 
     def get_linearized_discrete_dynamics(self, state, action):
         q = np.array([state[0], state[2], state[4]])
         v = np.array([state[1], state[3], state[5]])
-        lcs.set_linearization_point(q,v, action)
-        A, B_gen = lcs.get_A_matrix(), lcs.get_B_matrix()
-        pmat = lcs.get_P_matrix()
+        self.lcs.set_linearization_point(q,v, action)
+        A, B_gen = self.lcs.get_A_matrix(), self.lcs.get_B_matrix()
+        pmat = self.lcs.get_P_matrix()
         B = B_gen@pmat
-        C_mat = lcs.get_C_matrix()
-        G_mat = lcs.get_G_matrix()
-        J_mat = lcs.get_J_matrix()
-        H_mat = lcs.get_H_matrix()
+        C_mat = self.lcs.get_C_matrix()
```

Comment:
For all of these variables, you could more quickly get all of these values via `A, B, C, d, G, H, J, l, P = self.lcs.get_lcs_terms()`.  If you don't need some of those values, you can just replace the assignment with `_` (e.g. `A, B_gen, C_mat, _, G_mat, H_mat, J_mat, _, _ = self.lcs.get_lcs_terms()`).